docs/api.rst

```diff
@@ -24,7 +24,7 @@ Core
 
 .. autodata:: attr.NOTHING
 
-.. autofunction:: attr.s(these=None, repr_ns=None, repr=None, cmp=None, hash=None, init=None, slots=False, frozen=False, weakref_slot=True, str=False, auto_attribs=False, kw_only=False, cache_hash=False, auto_exc=False, eq=None, order=None, auto_detect=False, collect_by_mro=False, getstate_setstate=None, on_setattr=None)
+.. autofunction:: attr.s(these=None, repr_ns=None, repr=None, cmp=None, hash=None, init=None, slots=False, frozen=False, weakref_slot=True, str=False, auto_attribs=False, kw_only=False, cache_hash=False, auto_exc=False, eq=None, order=None, auto_detect=False, collect_by_mro=False, getstate_setstate=None, on_setattr=None, transform_hook=None)
```

Comment:
Do you think we could come up with a better name for the hook? While through the lens of this PR it makes complete sense, I feel its a bit generic next to the other options. It also bleeds out the name of the function that calls it which doesn't make much sense to the users. Would you agree?

Reply:
Naming things is so hard! 🤪 Is something like `attrib_converter()` better?

Reply:
So this is awkward.

The reason why new-style APIs use `field` is because it's quite distinct to the soup of attrs, attribs, attr and whatnot.

Given we're marching towards a brighter future…maybe `field_transformer`? 😇

Summoning the creative juices of @wsanchez @glyph @euresti @Tinche etc 😅

Reply:
+1 to `field_transformer`